tee-worker/cli/src/trusted_base_cli/commands/nonce.rs

```diff
@@ -30,8 +31,13 @@ pub struct NonceCommand {
 }
 
 impl NonceCommand {
-	pub(crate) fn run(&self, cli: &Cli, trusted_args: &TrustedCli) {
-		let who = get_pair_from_str(trusted_args, &self.account);
-		println!("{}", get_layer_two_nonce!(who, cli, trusted_args));
+	pub(crate) fn run(&self, cli: &Cli, trusted_cli: &TrustedCli) {
+		let (_mrenclave, shard) = get_identifiers(trusted_cli);
+		let who = get_pair_from_str(trusted_cli, &self.account);
+		let worker_api_direct = get_worker_api_direct(cli);
+		let nonce_ret = worker_api_direct.get_next_nonce(&shard, &(who.public().into()));
+		let nonce = nonce_ret.unwrap();
+		worker_api_direct.close().unwrap();
+		println!("{}", nonce);
```

Comment:
You can print `nonce` as soon as you get it (before closing `worker_api_direct` which may eventually fail)